packages/ui/src/main.ts

```diff
@@ -0,0 +1,110 @@
+import show, { type WalletProviderWithStoreVersion } from "./modal"
+import Bowser from "bowser"
+import sn, { type StarknetWindowObject } from "get-starknet-core"
+
+type StoreVersion = "chrome" | "firefox" | "edge"
+
+function getStoreVersionFromBrowser(): StoreVersion {
+  const browserName = Bowser.getParser(window.navigator.userAgent)
+    .getBrowser()
+    .name?.toLowerCase()
+  switch (browserName) {
+    case "firefox":
+      return "firefox"
+    case "microsoft edge":
+      return "edge"
+    default:
+      return "chrome"
+  }
+}
+
+export interface ConnectOptions {
+  modalMode?: "alwaysAsk" | "canAsk" | "neverAsk"
+  modalTheme?: "light" | "dark" | "system"
+  storeVersion?: StoreVersion
+}
+
+export interface DisconnectOptions {
+  clearDefaultWallet?: boolean
+}
+
+export const connect = async ({
+  modalMode = "canAsk",
+  storeVersion = getStoreVersionFromBrowser(),
+  modalTheme,
+}: ConnectOptions = {}): Promise<StarknetWindowObject | null> => {
```

Comment:
we'll need `| undefined` for backward compatibility (we can have both `| null | undefined`)